File: app/src/main/java/com/example/smarthomecontrol/GestureConstants.java
package com.example.smarthomecontrol;

import java.util.Collections;
import java.util.LinkedHashMap;
import java.util.Map;

/**
 * Maps gesture display names (shown in the Spinner) to their
 * corresponding gesture labels used for inter-activity communication.
 */
public final class GestureConstants {

    public static final Map<String, String> GESTURE_MAP;

    static {
        Map<String, String> map = new LinkedHashMap<>();
        map.put("Turn on lights", "LightOn");
        map.put("Turn off lights", "LightOff");
        map.put("Turn on fan", "FanOn");
        map.put("Turn off fan", "FanOff");
        map.put("Increase fan speed", "FanSpeedUp");
        map.put("decrease fan speed", "FanSpeedDown");
        map.put("Set Thermostat to specified temperature", "SetThermo");
        map.put("0", "Num0");
        map.put("1", "Num1");
        map.put("2", "Num2");
        map.put("3", "Num3");
        map.put("4", "Num4");
        map.put("5", "Num5");
        map.put("6", "Num6");
        map.put("7", "Num7");
        map.put("8", "Num8");
        map.put("9", "Num9");
        GESTURE_MAP = Collections.unmodifiableMap(map);
    }

    private GestureConstants() {
        // Prevent instantiation
    }
}

File: app/src/main/java/com/example/smarthomecontrol/GestureConstants.kt
package com.example.smarthomecontrol

object GestureConstants {

    const val EXTRA_GESTURE_LABEL = "EXTRA_GESTURE_LABEL"

    val gestureDisplayNames = listOf(
        "Turn on lights",
        "Turn off lights",
        "Turn on fan",
        "Turn off fan",
        "Increase fan speed",
        "Decrease fan speed",
        "Set Thermostat to specified temperature",
        "0", "1", "2", "3", "4", "5", "6", "7", "8", "9"
    )

    val gestureLabelMap = mapOf(
        "Turn on lights" to "LightOn",
        "Turn off lights" to "LightOff",
        "Turn on fan" to "FanOn",
        "Turn off fan" to "FanOff",
        "Increase fan speed" to "FanUp",
        "Decrease fan speed" to "FanDown",
        "Set Thermostat to specified temperature" to "Set Thermo",
        "0" to "Num0",
        "1" to "Num1",
        "2" to "Num2",
        "3" to "Num3",
        "4" to "Num4",
        "5" to "Num5",
        "6" to "Num6",
        "7" to "Num7",
        "8" to "Num8",
        "9" to "Num9"
    )
}
